### DISCUSSION TITLE
The MAUI team needs to provide a page for convincing why to choose MAUI over Flutter

### DISCUSSION THREAD

**Discussion by Happypig375** [dotnet/maui#238] [General]
If the team really cares about MAUI, then this needs to be done.

**Comment by zprima** [comment]
🤔 does it? I dont think so.

Instead of convicing, it should be something like "Coming from another platform?" and then a page for "from android to maui", "from ios to maui", "from flutter to maui".

**Reply by Happypig375** [reply]
Sure, ignoring competition does sound like a great way to succeed.

**Reply by zprima** [reply]
I dont see jetpack compose, swiftui, react native, flutter doing any convincing on their official pages. Instead, they give you explanations how their architecture works if you are coming from x platform to theirs. Which makes sense that MAUI should also have this "coming from x platform" section somewhere. Like they will have migration plans from XF to MAUI.

**Reply by Atreides07** [reply]
Perhaps this is needed not for users but for MAUI team. How can we understand why we should choose MAUI instead of Flutter, if even the MAUI team themselves do not know the benefits of MAUI over Flutter? "Because .net" is not enough today

**Reply by Atreides07** [reply]
To clarify: I'm a fan of Xamarin.Forms and have been working with Xamarin since Monodroid, Monotouch. But Flutter is pretty good. Very fast development speed and lower TimeToMarket than Xamarin. For me, Dart is a big drawback. It would be cool to have a Flutter analog with .net. Native UI is cool, but I don't really understand yet how MAUI can compare in development speed with Flutter, if again you have to spend a lot of time on custom controls on each platform.

**Reply by zprima** [reply]
Well, MAUI is not even in preview, nothing is finalized, even the final form of the syntax is up in the air. And they should convince us that is better then Flutter, based on what? There are no production apps, no sample apps, nothing. That will come much later in 2021 hopefully.

I'm trying to understand, why would the job of MAUI team be to convince you to use MAUI instead of Flutter, and not any other platform? Nobody from any framework does that directly. That would then imply its better then X but not better then Y. A community member can make "MAUI vs Flutter" comparison in a Youtube video, or a blog post "20 reasons why MAUI is better then X". That I understand. 

If by convincing you guys mean, that they should show how you can build X apps quickly and fast with MAUI, then yes, I agree. Everybody does that. When they showcase some sample apps, the arhitecture and the flow of how things are done, then the code and samples would do the convincing.

**Reply by Happypig375** [reply]
> A community member can make "MAUI vs Flutter" comparison in a Youtube video, or a blog post "20 reasons why MAUI is better then X". That I understand.

Currently, it's just really hard to do so. Maybe you can enumerate a few points.

**Reply by zprima** [reply]
Currently its near imposible to do so. Have to wait until Preview 1 is released. Then some feature preview posts should start appearing with each preview release of MAUI, maybe some "counter" tutorial apps or some basic ones. But until RC1 is out I don't expect many such video/blog posts, because everything is subject to change.

**Comment by bondarenkod** [comment]
There is no competition between MAUI & Flutter unless MAUI show us the same or the near the same performance. 
The performance is the main disadvantage of the XF.

**Comment by davidbuckleyni** [comment]
Why Flutter is just a good Ui designer tool its not great at full functional apps @bondarenkod  if your just on here to down tod MAUI and not a developer in the .net echo system why should they

**Comment by mrwcjoughin** [comment]
Flutters performance better than Xamarin.Forms? Are you kidding? Xamarin,
whether it’s forms or direct targeting used native controls in the end - so
you get a native app for the given platform. For that reason alone flutter
cannot be faster

On Fri, 16 Oct 2020 at 23:21, Dmytro Bondarenko <notifications@github.com>
wrote:

> There is no competition between MAUI & Flutter unless MAUI show us the
> same or the near the same performance.
> The performance is the main disadvantage of the XF.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dotnet/maui/discussions/238#discussioncomment-102459>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACOGERQQDA5L2J6GN5VVVLLSLC2MDANCNFSM4ROGHMUA>
> .
>
-- 
Thanks,

Matthew Joughin
Founder / CTO

+27 81 529 7169
http://www.adaptableapps.net

**Reply by Atreides07** [reply]
Have you tried to develop anything with Flutter? Xamarin.Forms is very slow. Many developers and project use Xamarin.Native only because Forms is very slow. But sometimes the Xamarin.Native performance is lacking. Interop with native controls is very expensive. Flutter doesn't need interop. It draws its controls directly. .MAUI will change the rendering engine due to the fact that Xamarin.Forms in its current form simply cannot be done faster

**Reply by saint4eva** [reply]
What did you mean by interop?

**Reply by SagarPanwala** [reply]
I would like to add that Flutter performance is far better than Xamarin Native.
Just try Hello world app in both platform and see start up time.
If you want more idea, try their Listtiles compare to Xamarin Listview for android and TableView for iOS without images, but try multiple item templates.
You will get clear idea.
Please don't directly deny anything if you haven't tried based on things that they are drawing so it can't be fast.

I'm using Xamarin since 2012, but I found flutter quite impressive with Design, Styling, Performance, but it lacks c# and xaml, if anychance it will get c# many developer will try it out for sure.

**Comment by mrwcjoughin** [comment]
Since when does Xamarin interop? Xamarin cross complies .net byte core to
objective-c for iOS and macOS and to Java for android - there is no interop
needed ???

On Sat, 17 Oct 2020 at 15:46, Akhmed Sheriev <notifications@github.com>
wrote:

> Have you tried to develop anything with Flutter? Xamarin.Forms is very
> slow. Many developers and project use Xamarin.Native only because Forms is
> very slow. But sometimes the Xamarin.Native performance is lacking. Interop
> with native controls is very expensive. Flutter doesn't need interop. It
> draws its controls directly. .MAUI will change the rendering engine due to
> the fact that Xamarin.Forms in its current form simply cannot be done faster
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/dotnet/maui/discussions/238#discussioncomment-102663>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACOGERULMLPJP5O6BOVCOF3SLGNZ7ANCNFSM4ROGHMUA>
> .
>
-- 
Thanks,

Matthew Joughin
Founder / CTO

+27 81 529 7169
http://www.adaptableapps.net

**Reply by Atreides07** [reply]
https://docs.microsoft.com/ru-ru/xamarin/android/deploy-test/release-prep/?tabs=macos#bundle-assemblies-into-native-code
_The AOT Compilation option requires an Enterprise license or higher._ 

AOT is a paid feature in Xamarin. 
Paid Xamarin is another disadvantage of Xamarin in choosing MAUI vs Flutter. (We used Rider instead of Visual Studio for Mac) Also, if you have an Enterprise license, you can compare the output apk size for Xamarin and Kotlin

**Reply by WashingtonARamos** [reply]
AOT & LLVM being an Enterprise feature is such a let down. They should at least let the Professional tier have it, Enterprise license is so much more expensive for quite literally this one feature if you're a small mobile dev using Xamarin.

**Comment by mrwcjoughin** [comment]
judging performance purely on startup time is absurd

**Reply by SagarPanwala** [reply]
Sorry to say, but I think you didn't read my whole message, I've already mentioned clearly that if you use ListView control then also you will find that Performance is far better.
I do like Xamarin Forms, but I want to get things better if they are stopping major release for 1 year.
But truth should be accepted, we can't just deny that they are drawing control , so performance must be poor, before we judge, we should try it.

**Comment by mrwcjoughin** [comment]
Xamarin.Forms does not draw controls? it explicitly uses the native
controls, except in rare cases on platforms like Linux where controls are
missing?

the listview is a mixed bag, specifically on Android, but with fast
renderers it's much better

**Comment by edsulaiman** [comment]
In my experience using Xamarin Forms, MAUI has had to make major changes in order to compete with other frameworks.

I have some problems that I have while using Xamarin Forms:
1. ListView performance is quite poor.
2. Everything is made to look native, sometimes our clients want apps which have the same UI on different platforms.
3. Difficulty customizing the UI.
4. Hot reload feature which is quite useless because it only responds to xaml changes.
5. Expensive

I've tried using React Native and Flutter, in terms of UI customization, they are far superior to Xamarin Forms.

**Comment by mrwcjoughin** [comment]
I agree on the learning curve with Xamarin.Forms is higher than the other
frameworks you mention.
However, the ability to share code with the backend is what makes me choose
XF/MAUI

In terms of expensive - what exactly are you referring to?
Visual Studio 2019 for both Windows and Mac is free in the community
edition?
I have used that for many years and don't miss any functionality?
